src/node_binding.cc

```diff
@@ -69,6 +69,7 @@
   V(task_queue)                                                                \
   V(tcp_wrap)                                                                  \
   V(timers)                                                                    \
+  V(tracing)                                                                   \
```

Comment:
Why do we need a new binding here? It seems this can just be part of the trace_events binding?

Reply:
The first reason was that I could not make it compile by putting the code inside the `node_trace_events.cc` to reuse `trace_events` bindings, but I think I can just change the namespace and the binding to `trace_events` on `node_tracing.cc`, I will try to do that.

Reply:
@joyeecheung I tried to remove the `tracing` binding but I don't know what I should do with: `NODE_BINDING_CONTEXT_AWARE_INTERNAL(
    trace_events, node::tracing::BindingData::CreatePerContextProperties)`, it gave me an error when because it's declared twice, on `node_tracing.cc` and on `node_trace_events.cc`.

I can remove the whole code of binding data and put the property initialization inside the `node_trace_events.cc` but I don't know if is appropriated to introduce that variable without support serialization.

Reply:
I did a test moving just the buffer declaration to `node_trace_events.cc`:

```c++
void NodeCategorySet::Initialize(Local<Object> target,
                Local<Value> unused,
                Local<Context> context,
                void* priv) {
  Environment* env = Environment::GetCurrent(context);
  Isolate* isolate = env->isolate();

  // Get the pointer of the memory for the flag that
  // store if trace is enabled for http the pointer will
  // always be the same and if the category does not exist
  // it creates: https://github.com/nodejs/node/blob/6bbf2a57fcf33266c5859497f8cc32e1389a358a/deps/v8/src/libplatform/tracing/tracing-controller.cc#L322-L342
  uint8_t* http_flag_pointer = const_cast<uint8_t*>(
    TRACE_EVENT_API_GET_CATEGORY_GROUP_ENABLED("node.http"));

  // NO_OP deleter
  // the idea of this chunk of code is to construct
  auto nop_deleter = [](void*, size_t, void*) {};
  auto backing_store = v8::ArrayBuffer::NewBackingStore(
    http_flag_pointer, 1, nop_deleter, nullptr);
  v8::Local<ArrayBuffer> array_buffer = v8::ArrayBuffer::New(
    isolate, std::move(backing_store));
  v8::Local<Uint8Array> uint8Array = v8::Uint8Array::New(array_buffer, 0, 1);

  target
      ->Set(context,
            FIXED_ONE_BYTE_STRING(isolate, "tracingCategories"),
            uint8Array)
      .Check();
```

In JS:

```
const trace_events = internalBinding('trace_events');

function isTraceHTTPEnabled() {
  return trace_events.tracingCategories[0] > 0;
}
```

And this didn't work, on my tests, the flag is not tracked anymore and the value in the array is always `0`.

So I think the `BindingData` must be used in this case.

About the `tracing` binding, we can remove if we find a solution to `NODE_BINDING_CONTEXT_AWARE_INTERNAL`.

Reply:
After learning more by doing another PR, I was able to move to `trace_events` file, so I think it should compile now.